build.rs

```diff
@@ -1,6 +1,65 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+use std::process::Command;
+use std::path::Path;
+
 fn main() {
     built::write_built_file().expect("Failed to acquire build-time information");
+    
+    // Build the frontend if it exists and Node.js is available
+    let frontend_dir = Path::new("frontend");
+    if frontend_dir.exists() {
+        println!("cargo:rerun-if-changed=frontend/src");
+        println!("cargo:rerun-if-changed=frontend/package.json");
+        
+        // Check if Node.js is available
+        if Command::new("node").arg("--version").output().is_ok() {
+            println!("Building frontend with Node.js...");
+            
+            // Install dependencies if node_modules doesn't exist
+            let node_modules = frontend_dir.join("node_modules");
+            if !node_modules.exists() {
```

Comment:
We should just run this every time. If we changed `package.json` this would need to run anyway. Otherwise it should just check cache